controllers/admin/AdminCarriersController.php

```diff
@@ -642,13 +642,13 @@ public function changeZones($id)
      */
     public function getList($id_lang, $order_by = null, $order_way = null, $start = 0, $limit = null, $id_lang_shop = false)
     {
-        parent::getList($id_lang, $order_by, $order_way, $start, $limit, $id_lang_shop);
+        // Replace "0" by the name of the shop directly in SQL query (allowing sort without errors)
+        $this->_select = 'CASE WHEN name = "0" '
```

Comment:
I think you need to do this:
```suggestion
        $this->_select .= ', CASE WHEN name = "0" '
```

... otherwise, you're replacing this: https://github.com/PrestaShop/PrestaShop/blob/30f668874fdc21d55eecb5a8fda6a48af317fe28/controllers/admin/AdminCarriersController.php#L138